query/executor.go

```diff
@@ -289,6 +292,98 @@ func (e *Executor) WithLogger(log *zap.Logger) {
 	e.TaskManager.Logger = e.Logger
 }
 
+func initQueryLogWriter(log *zap.Logger, e *Executor, path string) (*os.File, error) {
+	logFile, err := os.OpenFile(path, os.O_CREATE|os.O_RDWR|os.O_APPEND, 0644)
+	if err != nil {
+		e.Logger.Error("failed to open log file", zap.Error(err))
+		return nil, err
+	}
+
+	existingCore := log.Core()
+
+	encoderConfig := zap.NewProductionEncoderConfig()
+
+	fileCore := zapcore.NewCore(
+		zapcore.NewJSONEncoder(encoderConfig),
```

Comment:
Should we respect the [configuration setting for log format](https://docs.influxdata.com/enterprise_influxdb/v1/administration/configure/config-data-nodes/#format)?